crates/bevy_render/src/camera/camera.rs

```diff
@@ -233,6 +260,27 @@ impl Camera {
             None
         }
     }
+
+    /// Given a position in Normalized Device Coordinates,
+    /// use the camera's viewport to compute the world space position.
+    ///
+    /// When the position is within the viewport the values returned will be between -1.0 and 1.0 on the X and Y axes,
+    /// and between 0.0 and 1.0 on the Z axis.
+    /// To get the world space coordinates with the viewport position, you should use
+    /// [`world_to_viewport`](Self::world_to_viewport).
+    pub fn ndc_to_world(&self, camera_transform: &GlobalTransform, ndc: Vec3) -> Option<Vec3> {
+        // Build a transformation matrix to convert from NDC to world space using camera data
+        let ndc_to_world =
+            camera_transform.compute_matrix() * self.computed.projection_matrix.inverse();
+
+        let world_space_coords = ndc_to_world.project_point3(ndc);
+
+        if !world_space_coords.is_nan() {
+            Some(world_space_coords)
+        } else {
+            None
+        }
```

Comment:
Could be simplified with `bool::then`
```suggestion
        (!world_space_coords.is_nan()).then(|| world_space_coords)
```